README.md
# Lan Management System (LMS)
LMS (LAN Management System) is a package of applications for managing LAN networks. 
Its main goal is to provide the best service to customers, as seen in large ISP companies. 
LMS is written in PHP, Perl and C and can use MySQL or PostgreSQL as its database backends. 
The following features are provided at the time: customer database (names, addresses, phones, comments, etc),
computers inventory (IP, MAC), simple financial system suited for network operations, financial balances and invoices, email warnings to users, automatic billing schedule, ability to generate (almost) any kind of config file ie. ipchains/iptables firewall scripts, DHCP daemon configuration, zones for bind, /etc/ethers entries, oident, htb and more, visualization of bandwidth consumption per host, request tracker system (Helpdesk), timetable (Organizer).

This version is extend with lms-stck - module for (basic) small-store sales inluding multiple warehouse and (basic) stock management and has modified index.php file, which uses lms2.ini config file instead of lms.ini - change made for users to not to change main DB by accident


doc/ChangeLog
$Id$

version ?.??.?? (????-??-??):

  - userpanel enhancement: show confirmed documents only checkbox decides if confirmed
    documents are only shown for customer [chilan]
  - documentview enhancement: allow to view many pdf documents as single pdf file [chilan]
  - userpanel enhancement: module order can be changed from LMS userpanel configuration
    using drag&drop (fixes LMS+ #102) [chilan]
  - improvement - don't show boxes for contact types when no contact
    type is assigned to customer [chilan]
  - enhancement: phpui.disable_active_links_in_nodeinfo controls if some controls
    are active ui elements [chilan]
  - performance improvement: phpui.datatables_max_records limits datatables initialisation
    to tables with specified number of records (default: 300) [chilan]
  - improvement: click on tariff tag should add tag to filter and click on right side
    should move to tariff tag info (should fix LMS+ #101) [chilan]
  - added support for attribution network device for customer [kosa/chilan]
  - enhancement: added support for pro forma invoices (should resolve LMS+ #120) [chilan]
  - enhancement: support of convertion pro forma invoice to invoice [chilan]
  - improvement: splited user name to separated first name and last name components [kyob]
  - enhancement: support of customer number plans with special symbol %C [chilan]
  - enhancement: allow to select first free ip address automatically in node forms
    (phpui.first_reserved_addresses controls how many first addresses are omitted
    for selected network) [chilan]
  - added support for assignment tariffs to network device hosts [kosa/chilan]
  - improvement: if ip address is not empty go to the page in ip address selection
    window where this non-empty ip address is visible [chilan]
  - added support for set type of voip pool numbers [kosa/chilan]
  - improvement: allow to change number and number plan for edited invoices [chilan]
  - improvement: phpui.helpdesk_author_notify configuration variable controls
    if new helpdesk ticket/message/note notifications are sent to author [chilan]
  - improvement: allow to search for customers which are not assigned to any
    customer groups [chilan]
  - bugfix: taxes were inproperly displayed on invoice for both zero tax and tax free
    values were present at the same time [chilan]
  - enhancement: added node session support for ip addresses assigned to network
    devices [chilan]
  - enhancement: allow to automatically get first free address for network device
    ip addresses [chilan]
  - enhancement: allow to specify indefinite cutoff stop dates [chilan]
  - enhancement: JPK FA (invoices) support added [chilan]
  - enhancement: allow to define teryt addresses and tax office codes for divisions [chilan]
  - improvement: JPK FA (invoices) report uses recently added division teryt location
    and division tax office assigned [chilan]
  - improvement: allow to print web layout without left menu bar, top autosuggest fields
    and toolbar [chilan]
  - enhancement: show document creator on document lists [chilan]
  - enhancement: added support for confirmation user storage in documents [chilan]
  - enhancement: allow to select which helpdesk ticket details are printed [chilan]
  - enhancement: added document number sort cryterion to invoice report [chilan]
  - added script to load and bind buildings with teryt localizations [kosa/chilan]
  - improvement: switched to new multi attachment code for new mail messages [chilan]
  - improvement: when we try to send messages to more than 50 customers at once, confirmation
    dialog is displayed [chilan]
  - enhancement: cash source: added suport for main bank account definition [chilan]
  - enhancement: lms-notify.php: all actions are logged into transaction log [chilan]
  - improvement: voip.cdr_billing_record_format configuration variable describes
    format of billing record [chilan]
  - improvement: allow to print invoices manually issued [chilan]
  - enhacement: added documents flag to email and phone customer contacts which allows
    to decide if those contacts should be printed on documents [chilan]
  - enhancement: added support for effective dates of tariffs and promotions - only effective
    ones are displayed by default [chilan]
  - enhancement: added rbe info fields to divisions and rbe name for customers [chilan]
  - enhancement: allow filter document list by creator/authorising and all date types
    for periods [chilan]
  - moved addresses from nodes, netdevices, customers, netnodes, voipaccounts and divisions
    to one table [kosa/chilan]
  - removed possibility for setting up custom address for nodes and voipaccounts [kosa/chilan]
  - added possibility for create many addresses for each customer [kosa/chilan]
  - improvement: added new event types - vacation and duty [chilan]
  - enhancement: number plan list is updated automatically after customer or document
    type selection change [chilan]
  - improvement: sendinvoices.mail_format=text|html ('text' by default) allows
    to select mail format between html and text [chilan]
  - improvement: for new receipt show only cash registries to which cashier have
    read/write access privileges (we also take selected customer division into account) [chilan]
  - improvement: switched to pear/mail maintained by composer [chilan]
  - improvement: invoices: dynamic adjustment of number plan list according to
    selected customer division [chilan]
  - improvement: show db backup sizes also in human friendly units [chilan]
  - improvement: switched to xajax 0.6 beta fork from https://github.com/feuzeu/xajax which
    generates much less warnings in PHP 7 (if any at all) [chilan]
  - enhancement: added support for JPK-VAT exports [chilan]
  - rewrite lms-teryt to PHP [kosa/chilan]
  - merged lms-teryt and lms-teryt-buildings scripts [kosa/chilan]
  - added posibility to set recipient for invoices [kosa/chilan]

lib/definitions.php
<?php

/*
 * LMS version 1.11-git
 *
 *  (C) Copyright 2001-2016 LMS Developers
 *
 *  Please, see the doc/AUTHORS for more information about authors!
 *
 *  This program is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License Version 2 as
 *  published by the Free Software Foundation.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program; if not, write to the Free Software
 *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
 *  USA.
 *
 *  $Id$
 */

// that definitions should be included before LMS.class.php but after Smarty

// customers and contractor type
define('CTYPES_PRIVATE',0);
define('CTYPES_COMPANY',1);
define('CTYPES_CONTRACTOR',2);

$CTYPES = array(
    CTYPES_PRIVATE	=> trans('private person'),
    CTYPES_COMPANY	=> trans('legal entity'),
    CTYPES_CONTRACTOR	=> trans('contractor'),
);

// customer statuses
define('CSTATUS_INTERESTED', 1);
define('CSTATUS_WAITING', 2);
define('CSTATUS_CONNECTED', 3);
define('CSTATUS_DISCONNECTED', 4);
define('CSTATUS_DEBT_COLLECTION', 5);
define('CSTATUS_LAST', CSTATUS_DEBT_COLLECTION);

$CSTATUSES = array(
	CSTATUS_CONNECTED => array(
		'singularlabel' => trans('connected<!singular>'),
		'plurallabel' => trans('connected<!plural>'),
		'summarylabel' => trans('Connected:'),
		'img' => 'customer.gif',
		'alias' => 'connected'
	),
	CSTATUS_WAITING => array(
		'singularlabel' => trans('waiting'),
		'plurallabel' => trans('waiting'),
		'summarylabel' => trans('Waiting:'),
		'img' => 'wait.gif',
		'alias' => 'awaiting'
	),
	CSTATUS_INTERESTED => array(
		'singularlabel' => trans('interested<!singular>'),
		'plurallabel' => trans('interested<!plural>'),
		'summarylabel' => trans('Interested:'),
		'img' => 'unk.gif',
		'alias' => 'interested'
	),
	CSTATUS_DISCONNECTED => array(
		'singularlabel' => trans('disconnected<!singular>'),
		'plurallabel' => trans('disconnected<!plural>'),
		'summarylabel' => trans('Disconnected:<!summary>'),
		'img' => 'node_off.gif',
		'alias' => 'disconnected'
	),
	CSTATUS_DEBT_COLLECTION => array(
		'singularlabel' => trans('debt collection'),
		'plurallabel' => trans('debt collection'),
		'summarylabel' => trans('Debt Collection:<!summary>'),
		'img' => 'money.gif',
		'alias' => 'debtcollection'
	),
);

// Config types
define('CONFIG_TYPE_AUTO', 0);
define('CONFIG_TYPE_BOOLEAN', 1);
define('CONFIG_TYPE_POSITIVE_INTEGER', 2);
define('CONFIG_TYPE_EMAIL', 3);
define('CONFIG_TYPE_RELOADTYPE', 4);
define('CONFIG_TYPE_DOCTYPE', 5);
define('CONFIG_TYPE_MARGINS', 6);
define('CONFIG_TYPE_NONE', 7);
define('CONFIG_TYPE_RICHTEXT', 8);
define('CONFIG_TYPE_MAIL_BACKEND', 9);
define('CONFIG_TYPE_MAIL_SECURE', 10);
define('CONFIG_TYPE_DATE_FORMAT', 11);

$CONFIG_TYPES = array(
	CONFIG_TYPE_AUTO => trans('- auto -'),
	CONFIG_TYPE_NONE => trans('none'),
	CONFIG_TYPE_BOOLEAN => trans('boolean'),
	CONFIG_TYPE_POSITIVE_INTEGER => trans('integer greater than 0'),
	CONFIG_TYPE_EMAIL => trans('email'),
	CONFIG_TYPE_RELOADTYPE => trans('reload type'),
	CONFIG_TYPE_DOCTYPE => trans('document type'),
	CONFIG_TYPE_MARGINS => trans('margins'),
	CONFIG_TYPE_RICHTEXT => trans('visual editor'),
	CONFIG_TYPE_MAIL_BACKEND => trans('mail backend'),
	CONFIG_TYPE_MAIL_SECURE => trans('mail security protocol'),
	CONFIG_TYPE_DATE_FORMAT => trans('date format'),
);

// Helpdesk ticket status
define('RT_NEW', 0);
define('RT_OPEN', 1);
define('RT_RESOLVED', 2);
define('RT_DEAD', 3);

$RT_STATES = array(
    RT_NEW      => trans('new'),
    RT_OPEN     => trans('opened'),
    RT_RESOLVED => trans('resolved'),
    RT_DEAD     => trans('dead')
);

// Helpdesk cause type
define('RT_CAUSE_OTHER', 0);
define('RT_CAUSE_CUSTOMER', 1);
define('RT_CAUSE_COMPANY', 2);

$RT_CAUSE = array(
    RT_CAUSE_OTHER => trans("unknown/other"),
    RT_CAUSE_CUSTOMER => trans("customer's side"),
    RT_CAUSE_COMPANY => trans("company's side")
);

// Helpdesk note type
define('RTMESSAGE_REGULAR', 0);
define('RTMESSAGE_NOTE', 1);
define('RTMESSAGE_OWNER_CHANGE', 2);
define('RTMESSAGE_QUEUE_CHANGE', 4);
define('RTMESSAGE_STATE_CHANGE', 8);
define('RTMESSAGE_CAUSE_CHANGE', 16);
define('RTMESSAGE_CUSTOMER_CHANGE', 32);
define('RTMESSAGE_SUBJECT_CHANGE', 64);

// Messages status and type
define('MSG_NEW', 1);
define('MSG_SENT', 2);
define('MSG_ERROR', 3);
define('MSG_DRAFT', 4);
define('MSG_DELIVERED', 5);

define('MSG_MAIL', 1);
define('MSG_SMS', 2);
define('MSG_ANYSMS', 3);
define('MSG_WWW', 4);
define('MSG_USERPANEL', 5);
define('MSG_USERPANEL_URGENT', 6);

// Template types
define('TMPL_WARNING', 1);
define('TMPL_MAIL', 2);
define('TMPL_SMS', 3);
define('TMPL_WWW', 4);
define('TMPL_USERPANEL', 5);
define('TMPL_USERPANEL_URGENT', 6);

// Account types
define('ACCOUNT_SHELL', 1);
define('ACCOUNT_MAIL', 2);
define('ACCOUNT_WWW', 4);
define('ACCOUNT_FTP', 8);
define('ACCOUNT_SQL', 16);

// Document types
define('DOC_INVOICE', 1);
define('DOC_RECEIPT', 2);
define('DOC_CNOTE', 3);
//define('DOC_CMEMO', 4);
define('DOC_DNOTE', 5);
define('DOC_INVOICE_PRO',6);
define('DOC_INVOICE_PURCHASE',7);

define('DOC_CONTRACT', -1);
define('DOC_ANNEX', -2);
define('DOC_PROTOCOL', -3);
define('DOC_ORDER', -4);
define('DOC_SHEET', -5);
define('DOC_OTHER', -128);
define('DOC_BILLING',-10);
define('DOC_PRICELIST', -11);
define('DOC_PROMOTION', -12);
define('DOC_WARRANTY', -13);
define('DOC_REGULATIONS', -14);


$DOCTYPES = array(
    DOC_BILLING         =>      trans('billing'),
    DOC_INVOICE         =>      trans('invoice'),
    DOC_INVOICE_PRO     =>      trans('pro-forma invoice'),
    DOC_INVOICE_PURCHASE =>     trans('purchase invoice'),
    DOC_RECEIPT         =>      trans('cash receipt'),
    DOC_CNOTE       =>  trans('credit note'), // faktura korygujaca
//    DOC_CMEMO     =>  trans('credit memo'), // nota korygujaca
    DOC_DNOTE       =>  trans('debit note'), // nota obciazeniowa/debetowa/odsetkowa
    DOC_CONTRACT        =>      trans('contract'),
    DOC_ANNEX       =>  trans('annex'),
    DOC_PROTOCOL        =>      trans('protocol'),
    DOC_ORDER       =>  trans('order'),
    DOC_SHEET       =>  trans('customer sheet'), // karta klienta
    -6  =>      trans('contract termination'),
    -7  =>      trans('payments book'), // ksiazeczka oplat
    -8  =>      trans('payment summons'), // wezwanie do zapłaty
    -9  =>      trans('payment pre-summons'), // przedsądowe wezw. do zapłaty
    DOC_PRICELIST       =>  trans('price-list'), // cennik
    DOC_PROMOTION       =>  trans('promotion'), // promocja
    DOC_WARRANTY       =>  trans('warranty'), // gwarancja
    DOC_REGULATIONS       =>  trans('regulations'), // regulamin
    DOC_OTHER       =>  trans('other')
);

// Guarantee periods
$GUARANTEEPERIODS = array(
    -1 => trans('lifetime'),
    0  => trans('none'),
    12 => trans('$a months', 12),
    24 => trans('24 months', 24),
    36 => trans('$a months', 36),
    48 => trans('$a months', 48),
    60 => trans('$a months', 60)
);

// Internet Messengers
define('IM_GG', 0);
define('IM_YAHOO', 1);
define('IM_SKYPE', 2);

$MESSENGERS = array(
    IM_GG    => trans('Gadu-Gadu'),
    IM_YAHOO => trans('Yahoo'),
    IM_SKYPE => trans('Skype'),
);

define('DISPOSABLE', 0);
define('DAILY', 1);
define('WEEKLY', 2);
define('MONTHLY', 3);
define('QUARTERLY', 4);
define('YEARLY', 5);
define('CONTINUOUS', 6);
define('HALFYEARLY', 7);

// Accounting periods
$PERIODS = array(
    YEARLY	=>	trans('yearly'),
    HALFYEARLY  =>      trans('half-yearly'),
    QUARTERLY	=>	trans('quarterly'),
    MONTHLY	=>	trans('monthly'),
//    WEEKLY	=>	trans('weekly'),
//    DAILY	=>	trans('daily'),
    DISPOSABLE	=>	trans('disposable')
);

// Numbering periods
$NUM_PERIODS = array(
    CONTINUOUS	=>	trans('continuously'),
    YEARLY	=>	trans('yearly'),
    HALFYEARLY	=>	trans('half-yearly'),
    QUARTERLY	=>	trans('quarterly'),
    MONTHLY	=>	trans('monthly'),
//    WEEKLY	=>	trans('weekly'),
    DAILY	=>	trans('daily'),
);

// Tariff types
define('TARIFF_INTERNET', 1);
define('TARIFF_HOSTING', 2);
define('TARIFF_SERVICE', 3);
define('TARIFF_PHONE', 4);
define('TARIFF_TV', 5);
define('TARIFF_OTHER', -1);

// VoIP call types
define('CALL_INCOMING', 1);
define('CALL_OUTGOING', 2);

// VoIP call status
define('CALL_BUSY', 1);
define('CALL_ANSWERED', 2);
define('CALL_NO_ANSWER', 3);
define('CALL_SERVER_FAILED', 4);

// bit flags for VoIP call
define('CALL_FLAG_ADMIN_RECORDING', 1);
define('CALL_FLAG_CUSTOMER_RECORDING', 2);

$TARIFFTYPES = array(
	TARIFF_INTERNET	=> ConfigHelper::getConfig('tarifftypes.internet', trans('internet')),
	TARIFF_HOSTING	=> ConfigHelper::getConfig('tarifftypes.hosting', trans('hosting')),
	TARIFF_SERVICE	=> ConfigHelper::getConfig('tarifftypes.service', trans('service')),
	TARIFF_PHONE	=> ConfigHelper::getConfig('tarifftypes.phone', trans('phone')),
	TARIFF_TV	=> ConfigHelper::getConfig('tarifftypes.tv', trans('tv')),
	TARIFF_OTHER	=> ConfigHelper::getConfig('tarifftypes.other', trans('other')),
);

$PAYTYPES = array(
    1   => trans('cash'),
    2   => trans('transfer'),
    3   => trans('transfer/cash'),
    4   => trans('card'),
    5   => trans('compensation'),
    6   => trans('barter'),
    7   => trans('contract'),
    8   => trans('paid'),
);

// Contact types
define('CONTACT_MOBILE', 1);
define('CONTACT_FAX', 2);
define('CONTACT_LANDLINE', 4);
define('CONTACT_EMAIL', 8);
define('CONTACT_INVOICES', 16);
define('CONTACT_NOTIFICATIONS', 32);
define('CONTACT_BANKACCOUNT', 64);
define('CONTACT_TECHNICAL', 128);
define('CONTACT_URL', 256);
define('CONTACT_DISABLED', 16384);
define('CONTACT_DOCUMENTS', 32768);

$CONTACTTYPES = array(
    CONTACT_MOBILE          =>	trans('mobile'),
    CONTACT_FAX             =>	trans('fax'),
    CONTACT_INVOICES        =>	trans('invoices'),
    CONTACT_DISABLED        =>	trans('disabled'),
    CONTACT_NOTIFICATIONS   =>	trans('notifications'),
    CONTACT_TECHNICAL       =>	trans('technical'),
    CONTACT_IM_GG           =>	trans('Gadu-Gadu'),
    CONTACT_IM_YAHOO        =>	trans('Yahoo'),
    CONTACT_IM_SKYPE        =>	trans('Skype'),
    CONTACT_IM_FACEBOOK     =>	trans('Facebook'),
    CONTACT_DOCUMENTS		=>	trans('documents'),
);

define('DISCOUNT_PERCENTAGE', 1);
define('DISCOUNT_AMOUNT', 2);

$DISCOUNTTYPES = array(
	DISCOUNT_PERCENTAGE	=> '%',
	DISCOUNT_AMOUNT		=> trans('amount'),
);

define('DAY_MONDAY', 0);
define('DAY_TUESDAY', 1);
define('DAY_THURSDAY', 2);
define('DAY_WEDNESDAY', 3);
define('DAY_FRIDAY', 4);
define('DAY_SATURDAY', 5);
define('DAY_SUNDAY', 6);

$DAYS = array(
	DAY_MONDAY	=> trans('Mon'),
	DAY_TUESDAY	=> trans('Tue'),
	DAY_THURSDAY	=> trans('Thu'),
	DAY_WEDNESDAY	=> trans('Wed'),
	DAY_FRIDAY	=> trans('Fri'),
	DAY_SATURDAY	=> trans('Sat'),
	DAY_SUNDAY	=> trans('Sun'),
);

$LINKTYPES = array(
	0		=> trans('wire'),
	1		=> trans('wireless'),
	2		=> trans('fiber'),
);

$LINKTECHNOLOGIES = array(
	0 => array(
		1 => 'ADSL',
		2 => 'ADSL2',
		3 => 'ADSL2+',
		4 => 'VDSL',
		5 => 'VDSL2',
		10 => 'HDSL',
		11 => 'PDH',
		12 => 'POTS/ISDN',
		6 => '10 Mb/s Ethernet',
		7 => '100 Mb/s Fast Ethernet',
		8 => '1 Gigabit Ethernet',
		9 => '10 Gigabit Ethernet',
		50 => '(EURO)DOCSIS 1.x',
		51 => '(EURO)DOCSIS 2.x',
		52 => '(EURO)DOCSIS 3.x',
	),
	1 => array(
		100 => 'WiFi - 2,4 GHz',
		101 => 'WiFi - 5 GHz',
		102 => 'WiMAX',
		103 => 'LMDS',
		104 => 'radiolinia',
		105 => 'CDMA',
		106 => 'GPRS',
		107 => 'EDGE',
		108 => 'HSPA',
		109 => 'HSPA+',
		110 => 'DC-HSPA+',
		111 => 'MC-HSPA+',
		112 => 'LTE',
		113 => 'UMTS',
		114 => 'DMS',
	),
	2 => array(
		200 => 'CWDM',
		201 => 'DWDM',
		202 => 'SDH',
		203 => '10 Mb/s Ethernet',
		204 => '100 Mb/s Fast Ethernet',
		205 => '1 Gigabit Ethernet',
		206 => '10 Gigabit Ethernet',
		210 => '40 Gigabit Ethernet',
		207 => '100 Gigabit Ethernet',
		208 => 'EPON',
		209 => 'GPON',
		211 => 'ATM',
		212 => 'PDH',
		250 => '(EURO)DOCSIS 1.x',
		251 => '(EURO)DOCSIS 2.x',
		252 => '(EURO)DOCSIS 3.x',
	),
);

$LINKSPEEDS = array(
	10000		=> trans('10Mbit/s'),
	25000		=> trans('25Mbit/s'),
	54000		=> trans('54Mbit/s'),
	100000		=> trans('100Mbit/s'),
	200000		=> trans('200Mbit/s'),
	300000		=> trans('300Mbit/s'),
	1000000		=> trans('1Gbit/s'),
	10000000	=> trans('10Gbit/s'),
);

$BOROUGHTYPES = array(
	1 => trans('municipal commune'),
	2 => trans('rural commune'),
	3 => trans('municipal-rural commune'),
	4 => trans('city in the municipal-rural commune'),
	5 => trans('rural area to municipal-rural commune'),
	8 => trans('estate in Warsaw-Centre commune'),
	9 => trans('estate'),
);

$PASSWDEXPIRATIONS = array(
	0	=> trans('never expires'),
	7	=> trans('week'),
	14	=> trans('2 weeks'),
	21	=> trans('21 days'),
	31	=> trans('month'),
	62	=> trans('2 months'),
	93	=> trans('quarter'),
	183	=> trans('half year'),
	365	=> trans('year'),
);

$NETELEMENTSTATUSES = array(
	0	=> trans('existing'),
	1	=> trans('under construction'),
	2	=> trans('planned'),
);

$NETELEMENTTYPES = array(
	0	=> 'budynek biurowy',
	2	=> 'budynek mieszkalny',
	1	=> 'budynek przemysłowy',
	11	=> 'budynek usługowy',
	12	=> 'budynek użyteczności publicznej',
	3	=> 'obiekt sakralny',
	13	=> 'obiekt sieci elektroenergetycznej',
	5	=> 'wieża',
	4	=> 'maszt',
	10	=> 'komin',
	6	=> 'kontener',
	7	=> 'szafa uliczna',
	14	=> 'słup',
	8	=> 'skrzynka',
	9	=> 'studnia kablowa',
);

$NETELEMENTOWNERSHIPS = array(
	0	=> 'Węzeł własny',
	1	=> 'Węzeł współdzielony z innym podmiotem',
	2	=> 'Węzeł obcy',
);

$USERPANEL_ID_TYPES = array(
	1	=> array(
		'label' => trans('Customer ID:'),
		'selection' => trans('Customer ID and PIN'),
	),
	2	=> array(
		'label' => trans('Phone number:'),
		'selection' => trans('Phone number and PIN'),
	),
	3	=> array(
		'label' => trans('Document number:'),
		'selection' => trans('Document number and PIN'),
	),
	4	=> array(
		'label' => trans('Customer e-mail:'),
		'selection' => trans('Customer e-mail and PIN'),
	),
);

define('EVENT_OTHER', 1);
define('EVENT_NETWORK', 2);
define('EVENT_SERVICE', 3);
define('EVENT_INSTALLATION', 4);
define('EVENT_MEETING', 5);
define('EVENT_VACATION', 6);
define('EVENT_DUTY', 7);

$EVENTTYPES = array(
	EVENT_SERVICE      => trans('service<!event>'),
	EVENT_INSTALLATION => trans('installation'),
	EVENT_MEETING => trans('meeting'),
	EVENT_VACATION => trans('vacation'),
	EVENT_DUTY => trans('duty')
);

$EVENTSTYLES = array(
	EVENT_OTHER => 'background-color: gray; color: white;',
	EVENT_NETWORK => 'background-color: blue; color: white;',
	EVENT_SERVICE => 'background-color: red; color: white;',
	EVENT_INSTALLATION => 'background-color: green; color: white;',
	EVENT_MEETING => 'background-color: yellow; color: black;',
	EVENT_VACATION => 'background-color: white; color: black;',
	EVENT_DUTY => 'background-color: brown; color: white;'
);

define('SESSIONTYPE_PPPOE', 1);
define('SESSIONTYPE_DHCP', 2);
define('SESSIONTYPE_EAP', 4);
define('SESSIONTYPE_WIFI', 8);
define('SESSIONTYPE_VOIP', 16);

$SESSIONTYPES = array(
	SESSIONTYPE_PPPOE => array(
		'label' => trans('PPPoE Client'),
		'tip' => 'Enable/disable PPPoE Server Client'
	),
	SESSIONTYPE_DHCP => array(
		'label' => trans('DHCP Client'),
		'tip' => 'Enable/disable DHCP Server Client'
	),
	SESSIONTYPE_EAP => array(
		'label' => trans('EAP Client'),
		'tip' => 'Enable/disable EAP Server Client'
	),
	SESSIONTYPE_WIFI => array(
		'label' => trans('WiFi AP Client'),
		'tip' => 'Enable/disable WiFi AP Client access'
	),
	SESSIONTYPE_VOIP => array(
		'label' => trans('VoIP Gateway'),
		'tip' => 'Enable/disable VoIP Gateway access'
	),
);

if(isset($SMARTY))
{
	$SMARTY->assign('_CTYPES',$CTYPES);
	$SMARTY->assign('_CSTATUSES', $CSTATUSES);
	$SMARTY->assign('_DOCTYPES', $DOCTYPES);
	$SMARTY->assign('_PERIODS', $PERIODS);
	$SMARTY->assign('_GUARANTEEPERIODS', $GUARANTEEPERIODS);
	$SMARTY->assign('_NUM_PERIODS', $NUM_PERIODS);
	$SMARTY->assign('_RT_STATES', $RT_STATES);
	$SMARTY->assign('_CONFIG_TYPES', $CONFIG_TYPES);
	$SMARTY->assign('_TARIFFTYPES', $TARIFFTYPES);
	$SMARTY->assign('_PAYTYPES', $PAYTYPES);
	$SMARTY->assign('_CONTACTTYPES', $CONTACTTYPES);
	$SMARTY->assign('_DISCOUNTTYPES', $DISCOUNTTYPES);
	$SMARTY->assign('_DAYS', $DAYS);
	$SMARTY->assign('_LINKTYPES', $LINKTYPES);
	$SMARTY->assign('_LINKTECHNOLOGIES', $LINKTECHNOLOGIES);
	$SMARTY->assign('_LINKSPEEDS', $LINKSPEEDS);
	$SMARTY->assign('_BOROUGHTYPES', $BOROUGHTYPES);
	$SMARTY->assign('_PASSWDEXPIRATIONS', $PASSWDEXPIRATIONS);
	$SMARTY->assign('_NETELEMENTSTATUSES', $NETELEMENTSTATUSES);
	$SMARTY->assign('_NETELEMENTTYPES', $NETELEMENTTYPES);
	$SMARTY->assign('_NETELEMENTOWNERSHIPS', $NETELEMENTOWNERSHIPS);
	$SMARTY->assign('_USERPANEL_ID_TYPES', $USERPANEL_ID_TYPES);
	$SMARTY->assign('_EVENTTYPES', $EVENTTYPES);
	$SMARTY->assign('_EVENTSTYLES', $EVENTSTYLES);
	$SMARTY->assign('_SESSIONTYPES', $SESSIONTYPES);
}

define('DEFAULT_NUMBER_TEMPLATE', '%N/LMS/%Y');

// Investment project types
define('INV_PROJECT_REGULAR', 0);
define('INV_PROJECT_SYSTEM', 1);

// Address types
define('POSTAL_ADDRESS'          , 0);
define('BILLING_ADDRESS'         , 1);
define('LOCATION_ADDRESS'        , 2);
define('DEFAULT_LOCATION_ADDRESS', 3);
define('RECIPIENT_ADDRESS'       , 4);

?>
